From a7da2b1f63d792e40dc52e984cd97f379152b5d4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Mon, 4 Aug 2025 09:12:58 +0200
Subject: [PATCH] fix(meson): fix `gexiv2-0.16` lookup to respect `gexiv2`
 option
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Pass both `gexiv2-0.16` and `gexiv2` packages to a single `dependency()`
lookup, in order to simplify the logic and fix respecting `gexiv2`
option.  This is supported since meson 0.60.0.  The minimum version
can be left as `>=0.14`, since `meson-0.16` will always be newer than
`>=0.15`.  Thanks to Eli Schwartz for the suggestion.

Fixes #1522

Signed-off-by: Michał Górny <mgorny@gentoo.org>
---
 meson.build | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index 8887b3e8..4a0dfe94 100644
--- a/meson.build
+++ b/meson.build
@@ -82,10 +82,7 @@ clapper_gtk_dep = dependency('clapper-gtk-0.0', required: get_option('clapper'))
 gstreamer_dep = dependency('gstreamer-1.0', required: get_option('gstreamer'))
 webkit_dep = dependency('webkitgtk-6.0', required: get_option('in-app-browser'))
 
-gexiv2_dep = dependency('gexiv2-0.16', version: '>=0.15', required: false)
-if not gexiv2_dep.found ()
-  gexiv2_dep = dependency('gexiv2', version: '>=0.14', required: get_option('gexiv2'))
-endif
+gexiv2_dep = dependency('gexiv2-0.16', 'gexiv2', version: '>=0.14', required: get_option('gexiv2'))
 
 if not libwebp_dep.found ()
   warning('WebP support might be missing, please install webp-pixbuf-loader.')
